R/stats-factanal-tidiers.R

```diff
@@ -75,25 +106,35 @@ augment.factanal <- function(x, data, ...) {
   }
 
   # Place relevant values into a tidy data frame
-  tidy_df <- data.frame(.rownames = rownames(scores), data.frame(scores)) %>%
-    as_tibble()
+  if (has_rownames(scores)) {
+    tidy_df <- data.frame(.rownames = rownames(scores), data.frame(scores)) %>%
+      as_tibble() %>%
+      dplyr::mutate(.rownames = as.character(.rownames))
+  } else {
+    tidy_df <- tibble::rownames_to_column(as.data.frame(scores), var = ".rownames") %>% 
+      as_tibble() %>%
+      dplyr::mutate(.rownames = as.character(.rownames))
+  }
 
   colnames(tidy_df) <- gsub("Factor", ".fs", colnames(tidy_df))
 
   # Check if original data provided
   if (missing(data)) {
     return(tidy_df)
+  } else {
+    data <- tibble::rownames_to_column(as.data.frame(data), var = ".rownames") %>% 
+      as_tibble() %>%
+      dplyr::mutate(.rownames = as.character(.rownames))
   }
 
   # Bind to data
-  data$.rownames <- rownames(data)
   tidy_df <- tidy_df %>% 
     dplyr::right_join(data, by = ".rownames")
 
   tidy_df %>% 
     dplyr::select(
-      .rownames, everything(),
-      -matches("\\.fs[0-9]*"), matches("\\.fs[0-9]*")
+      .rownames, dplyr::everything(),
+      -dplyr::matches("\\.fs[0-9]*"), dplyr::matches("\\.fs[0-9]*")
```

Comment:
Perhaps comment on what the regex does to make it easier to maintain this down the line?